### ISSUE TITLE
is not under 'rootDir' 'src/'. even with exclude

### ISSUE THREAD

**Issue by cloudrex** [Microsoft/TypeScript#30162]
`exclude` compiler option is not working at all! I'm specifying `exclude` to the folder `gen` (and even `./gen`) and still receiving the error below.

**TypeScript Version:**  3.3.3333

**Search Terms:**

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
```

**Expected behavior:**
No errors.

**Actual behavior:**
```
error TS6059: File '.../ts-lang/gen/grammar/MyLangLexer.ts' is not under 'rootDir' 'src/'. 'rootDir' is expected to contain all source files.
```

**Playground Link:** 

**Related Issues:**

**Comment by j-oliveras** [comment]
Some of your files references it. Try setting traceResolution to true.

**Comment by RyanCavanaugh** [comment]
Your program is misconfigured, but the error could be more helpful.

When module resolution pulls in a file from outside a specified `rootDir` we should issue the error *on that import* instead of complaining way later in the process when we no longer know why that file is in the program.
